ow_plexil/rqt_plexil_plan_selection/src/rqt_plexil_plan_selection/rqt_plexil_plan_selection.py

```diff
@@ -180,24 +180,8 @@ def handle_send_button_clicked(self, checked):
     '''When send button is clicked any items in the plan queue table are sent (in order) to the sent plans table.
      It then publishes a string array of these plans so that the ow_plexil_plan_selection node can run them 
      sequentially. If the subscriber is not connected a popup box reminding the user to run the ow_plexil node will show up.'''
-    
-    #checks to see if we have connected to service yet
-    if(self.first_time == True):
-      #we get list of services and see if any match plexil_plan_selection
-      services = rosservice.get_service_list()
-      service_running = [i for i in services if "plexil_plan_selection" in i]
-      #if none exist we send a popup and return
-      if(len(service_running) == 0):
-        popup = QMessageBox()
-        popup.setWindowTitle("ow_plexil service not yet connected")
-        popup.setText("ow_plexil plan selection service not connected yet, please make sure the ow_plexil launch file is running.")
-        send_popup = popup.exec_()
-        return
-      else:
-        #client setup
-        rospy.wait_for_service('plexil_plan_selection')
-        self.plan_select_client = rospy.ServiceProxy('plexil_plan_selection', PlanSelection)
-        self.first_time = False
+    if(self.check_client_set_up() == False):
```

Comment:
You keep adding these extra braces to the python if statements. While they do no harm, they don't like nice in the eyes of a python developer.